### ISSUE TITLE
RLCS Season 1 and Season 2 LAN Grandfinals fail to parse

### ISSUE THREAD

**Issue by enzanki-ars** [tfausak/rattletrap#85]
I assume the following error is because there wasn't an official LAN build until later.  

```
could not get class name for object Str {strValue = "Archetypes.GameEvent.GameEvent_SoccarLan"}
```

These files are hosted at:

- Season 1 is only available contained within a zip folder.  They react the same as the Season 2 ones.
  - [X] S1 - Day 1 - https://drive.google.com/drive/folders/0BzFE4BT_MORAUXU1M09LQU9CNkU
  - [X] S1 - Day 2 - https://drive.google.com/drive/folders/0BzFE4BT_MORAY3lVNkJ1cHFYeWc
- Season 2
  - [ ]  S2 - Day 1 - https://drive.google.com/drive/folders/0BzFE4BT_MORAWVRLdGdWbnFuX2c
  - [ ]  S2 - Day 2 - https://drive.google.com/drive/folders/0BzFE4BT_MORATmRFWXNtVXZGX3c

**Comment by jjbott** [comment]
Yeah I saw this and fixed mine quick. :)

**Comment by enzanki-ars** [comment]
Based on https://github.com/jjbott/RocketLeagueReplayParser/commit/701846ecf72483f0463732fd71950c309cbbc286, I added a pull request to make a similar change (#86) as it was pretty easy.  Whith this change, every single RLCS replay parses correctly.

**Comment by enzanki-ars** [comment]
The season 2 ones still fail.  These ones fail with any one of the following errors:

```
rattletrap.exe: could not get name for index Word32le {word32leValue = 29}
rattletrap.exe: could not get name for index Word32le {word32leValue = 55}
rattletrap.exe: could not get name for index Word32le {word32leValue = 73}
rattletrap.exe: could not get name for index Word32le {word32leValue = 91}
```

This one is a little above my skill though....

**Comment by jjbott** [comment]
Oh man. I didn't try the S2 ones on mine, so I may have the same issue. I may not be able to check until tomorrow though.

**Comment by tfausak** [comment]
Weird! The "could not get name for index" error comes from here:

https://github.com/tfausak/rattletrap/blob/8c475b00e8063d42a86add8902f9d8bc8935161f/library/Rattletrap/Decode/SpawnedReplication.hs#L52-L58

I'm not sure why the RLCS S2 LAN replays would have name indexes that don't exist.

**Comment by jjbott** [comment]
This fixes parsing the S2 replays, but breaks a lot of others. Need to find a good key to enable/disable that bit of code: https://github.com/jjbott/RocketLeagueReplayParser/commit/04959fa11ecb066cb69d1c962082aaeaa57fd16d

**Comment by enzanki-ars** [comment]
I would assume that the version that these RLCS replays is unique compared to all of the other replays.  I wonder if you could just explicitly tell it to ignore the name for the index specifically for this version...?

**Comment by enzanki-ars** [comment]
Shoot.  The version reported is 868.14, same as many others that need that code...

**Comment by enzanki-ars** [comment]
Unless the RL client has logic to attempt decoding it and if that does not work attempt continuing, I don't see any better way.

**Comment by tfausak** [comment]
I cobbled a fix together: #88.

**Comment by tfausak** [comment]
After merging #88, Rattletrap was able to parse all of the replays in those Google Drive folders. The test suite continues to pass too, so I don't think I broke any other replays. 🤞
